### DISCUSSION TITLE
Conflict with `data-mantine-color-scheme` Attribute

### DISCUSSION THREAD

**Discussion by Elvincth** [mantinedev/mantine#6573] [Q&A]
We are encountering a problem with libraries like [jotai-devtools](https://github.com/jotaijs/jotai-devtools) that use Mantine v7. When a user uses the Mantine library, it injects the `[data-mantine-color-scheme]` attribute into the `<html>` tag. This overrides the CSS selectors in jotai-devtools, which also use the `[data-mantine-color-scheme]` attribute.

To resolve this issue, we request an option to change the `data-mantine-color-scheme` attribute to prevent conflicts with other libraries.

Thank you for your consideration.

**Comment by rtivital** [comment]
It is not possible to implement this feature. `data-mantine-color-scheme` attribute is used as a static value in css files. If you want to change the element where the attribute is applied – https://mantine.dev/theming/mantine-provider/#getrootelement

**Reply by Elvincth** [reply]
The issue we're encountering is that when using Mantine components alongside Jotai Devtools (which also relies on the Mantine component library), the `data-mantine-color-scheme` attribute in the HTML is being overridden. This occurs even after we've attempted to change the root element. We’ve tried various methods to manage CSS layering and scoping, similar to the approach described in [this pull request](https://github.com/jotaijs/jotai-devtools/pull/157). While this approach helped address CSS conflicts, it didn’t resolve the issue with the conflicting `data-mantine-color-scheme` attribute. 

Do you have any suggestions for resolving this issue?

Thanks!

**Reply by rtivital** [reply]
You can ask jotai to change root element on their side

**Reply by Elvincth** [reply]
Jotai already using another root element `span#jotai-devtools-root`  But the issue we have is that the host `data-mantine-color-scheme` attr that lives in the `html` tag overrides jotai's  root element `data-mantine-color-scheme` attr, due to css specificity. 
<img width="913" alt="image" src="https://github.com/user-attachments/assets/f4db0e5f-232e-420f-86c3-4c4804baee3d">

**Reply by rtivital** [reply]
I'm not sure what the issue then

**Reply by Elvincth** [reply]
> I'm not sure what the issue then

If a library using mantine and the project itself using mantine will have conflicts on the color scheme attr staring from v7

**Reply by arjunvegda** [reply]
Gentle bump @rtivital ^ Could we make Mantine more friendly for library authors? 🙏
